src/sentry/integrations/vercel/integration.py

```diff
@@ -165,38 +165,58 @@ def update_organization_config(self, data):
             if mapping in old_mappings:
                 continue
             [sentry_project_id, vercel_project_id] = mapping
-
             sentry_project = Project.objects.get(id=sentry_project_id)
             enabled_dsn = ProjectKey.get_default(project=sentry_project)
             if not enabled_dsn:
                 raise IntegrationError("You must have an enabled DSN to continue!")
+            source_code_provider = self.get_source_code_provider(vercel_client, vercel_project_id)
+            if not source_code_provider:
+                raise IntegrationError(
+                    "You must connect your Vercel project to a Git repository to continue!"
+                )
             sentry_project_dsn = enabled_dsn.get_dsn(public=True)
+            uuid = uuid4().hex
 
-            org_secret = self.create_secret(
-                vercel_client, vercel_project_id, "SENTRY_ORG", sentry_project.organization.slug
+            sentry_auth_token = SentryAppInstallationForProvider.objects.get(
+                organization=sentry_project.organization.id, provider="vercel"
             )
-            project_secret = self.create_secret(
-                vercel_client,
-                vercel_project_id,
-                "SENTRY_PROJECT_%s" % sentry_project_id,
+            secret_names = [
+                "SENTRY_ORG_%s" % uuid,
+                "SENTRY_PROJECT_%s" % uuid,
+                "NEXT_PUBLIC_SENTRY_DSN_%s" % uuid,
```

Comment:
@ceorourke Since we are using a uuid now, you can simplify the logic in `create_secret` to not have to make a `GET` request, we can assume the secrets are always unique.